### PR TITLE
Sets OpenStack LoadBalancer manage-security-groups setting

### LINKED ISSUE #1717: Automatically manage OpenStack LoadBalancer Security Groups in customer clusters
**User Story**
Now that https://github.com/kubernetes/kubernetes/pull/65373 has been merged and released in Kubernetes 1.9.10, 1.10.6 and 1.11.1, we would like to enable in our customer clusters that the LB security groups are managed automatically.

For that this configuration:

```
[LoadBalancer]
manage-security-groups = true
```

needs to be added to the customer cluster's cloud-config.

Unfortunately this configuration should not be added for older Kubernetes versions, since this would break the creation of an OpenStack LoadBalancer resource completely.
If it is too complicated to differentiate between versions, we would be also fine to add this setting in general and disable older Kubernetes versions.

**Acceptance criteria**

Configuration above is automatically added to all customer clusters running on OpenStack with Kubernetes versions >= 1.9.10, 1.10.6, 1.11.1.

### PR DESCRIPTION
**Pull request by bashofmann**
**What this PR does / why we need it**:

Sets OpenStack LoadBalancer manage-security-groups setting in cloud-config for Kubernetes versions that support it

The issue that this setting is not working was fixed with https://github.com/kubernetes/kubernetes/pull/65373 and
is available in versions >=1.9.10, 1.10.6, 1.11.1

The setting should not be applied to older Kubernetes versions since it breaks creating LoadBalancers completely.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1717

**Special notes for your reviewer**:

**Release note**:

```release-note cloud provider
OpenStack LoadBalancer manage-security-groups setting is set into cluster's cloud-config for Kubernetes versions where https://github.com/kubernetes/kubernetes/issues/58145 is fixed.
```

### CODE REVIEW

**Comment by bashofmann** [comment]
What I was unsure about: Should we make this configurable with a config flag or over the datacenters.yaml, so that some providers can activate/deactivate it in their setup?

File: api/pkg/resources/cloudconfig/configmap.go [hunk 1]
```diff
@@ -88,6 +88,28 @@ ignore-volume-az = {{ .DC.Spec.Openstack.IgnoreVolumeAZ }}
...
+
+[LoadBalancer]
+manage-security-groups = true
+{{- end }}{{- end }}{{- end }}
+{{- if eq (index $version "_0") "1" }}{{- if ge (int (index $version "_1")) 12 }}
+
+[LoadBalancer]
+manage-security-groups = true
+{{- end }}{{- end }}
+{{- end }}
```
Comment (mrIncompetent) [inline api/pkg/resources/cloudconfig/configmap.go]: Using the sprig semver functions makes this a bit easier:
```
{{- if .Cluster.Spec.Cloud.Openstack }}
[Global]
auth-url = "{{ .DC.Spec.Openstack.AuthURL }}"
username = "{{ .Cluster.Spec.Cloud.Openstack.Username }}"
password = "{{ .Cluster.Spec.Cloud.Openstack.Password }}"
domain-name= "{{ .Cluster.Spec.Cloud.Openstack.Domain }}"
tenant-name = "{{ .Cluster.Spec.Cloud.Openstack.Tenant }}"
region = "{{ .DC.Spec.Openstack.Region }}"

[BlockStorage]
trust-device-path = false
bs-version = "v2"
{{- if semverCompare ">=1.9.*" .Cluster.Spec.Version }}
ignore-volume-az = {{ .DC.Spec.Openstack.IgnoreVolumeAZ }}
{{- end }}

[LoadBalancer]
{{- if semverCompare "~1.9.10 || ~1.10.6 || ~1.11.1" .Cluster.Spec.Version }}
manage-security-groups = true
{{- end }}
{{- end }}
```

Comment (bashofmann) [reply api/pkg/resources/cloudconfig/configmap.go]: Nice, hadn't seen that. I'll change it like this.

Should I also make this configurable in the datacenters.yaml then?

Comment (mrIncompetent) [reply api/pkg/resources/cloudconfig/configmap.go]: Can always be set

**Review by mrIncompetent — DISMISSED** [review]

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**